cloud/previews/previews_test.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2024 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+package previews_test
+
+import (
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/uuid"
+	"github.com/madlambda/spells/assert"
+	"github.com/terramate-io/terramate/cloud"
+	"github.com/terramate-io/terramate/cloud/previews"
+	"github.com/terramate-io/terramate/config"
+	"github.com/terramate-io/terramate/errors"
+	"github.com/terramate-io/terramate/project"
+)
+
+func TestCreatePreview(t *testing.T) {
+	type want struct {
+		numStacksReturned   int
+		httpEndpointsCalled []string
+		err                 error
+	}
+	type testcase struct {
+		name           string
+		numRunContexts int
+		orgUUID        string
+		want           want
+	}
+
+	testcases := []testcase{
+		{
+			name:           "zero run contexts",
+			orgUUID:        "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee",
+			numRunContexts: 0,
+			want: want{
+				numStacksReturned:   0,
+				httpEndpointsCalled: []string{},
+				err:                 errors.E("no affected stacks or runs provided"),
+			},
+		},
+		{
+			name:           "empty org uuid returns an error",
+			orgUUID:        "",
+			numRunContexts: 3,
+			want: want{
+				numStacksReturned:   0,
+				httpEndpointsCalled: []string{},
+				err:                 errors.E("org uuid is empty"),
+			},
+		},
+		{
+			name:           "three stacks",
+			orgUUID:        "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee",
+			numRunContexts: 3,
+			want: want{
+				numStacksReturned: 3,
+				httpEndpointsCalled: []string{
+					"POST /v1/previews/aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee",
+				},
+				err: nil,
+			},
+		},
+	}
```

Comment:
only the *three stacks* is testing a real case. 
The first is a valid case (should not error), the second never happens because cli ensures the organization is selected. 

I think this testing approach is brittle and is not testing anything..
better do it later with the `testserver` or `interop` package.